### PR TITLE
Change: No need to save AI/GS data when joining a multiplayer server

### PR DESCRIPTION
**Pull request by SamuXarick**
#6478

### CODE REVIEW

File: src/saveload/saveload.cpp [hunk 1]
```diff
@@ -63,6 +63,7 @@ SaveLoadVersion _sl_version; ///< the major savegame version identifier
 byte   _sl_minor_version;    ///< the minor savegame version, DO NOT USE!
 char _savegame_format[8];    ///< how to compress savegames
 bool _do_autosave;           ///< are we doing an autosave at the moment?
+bool _save_empty_script;     ///< are we sending a map to a client over the nework?
```
Comment (PeterN) [inline src/saveload/saveload.cpp]: Comment and variable name are... what?

File: src/ai/ai_core.cpp [hunk 2]
```diff
@@ -281,7 +282,7 @@
 
 /* static */ void AI::Save(CompanyID company)
 {
-	if (!_networking || _network_server) {
+	if (!_networking || _network_server && !_save_empty_script) {
```
Comment (PeterN) [inline src/ai/ai_core.cpp]: Additional parenthesis required.

**Comment by TrueBrain** [comment]
Okay, correct me if I am wrong:
In 2016 you tried this PR, and it got declined with: this PR is fixing the wrong issue.
Without any argumentation why things are different now, you file the exact same PR again. I see no reason to not close it with the same argumentation.

In general, it is very impolite to try the same thing again, expecting a different outcome, without arguing your case. Please consider making your case, instead of just trying again.

**Comment by SamuXarick** [comment]
Because clients do not load AI/GS.
https://github.com/OpenTTD/OpenTTD/blob/8139b14e9c941433bce74e46a7330123cae4ae2b/src/saveload/ai_sl.cpp#L76-L79
https://github.com/OpenTTD/OpenTTD/blob/8139b14e9c941433bce74e46a7330123cae4ae2b/src/saveload/game_sl.cpp#L69-L73

**Pull request closed without merging**